provider/pkg/provider/provider.go

```
@@ -308,7 +309,14 @@ func (k *azureNativeProvider) Invoke(ctx context.Context, req *rpc.InvokeRequest
 		if err != nil {
 			return nil, err
 		}
-		body := crud.PrepareAzureRESTBody(id, parameters, nil, args.Mappable(), k.converter)
+		body, err := crud.PrepareAzureRESTBody(id, parameters, nil, args.Mappable(), k.converter)
+		if err != nil {
+			if body == nil {
+				return nil, fmt.Errorf("error preparing body for %s: %v", label, err)
+			}
+			// Log the error and continue with whatever was convertible.
```

Why continue instead of failing? Doesn't this risk creating hard-to-debug and reproduce errors?

---

We don't currently throw any errors from this method, so there might be cases where this is being hit right now but it silently continues. This just adds visibility of the issue for now, but leaves the behaviour unchanged (apart from if there's no body to continue with).